oar/core/shipment.py

```diff
@@ -0,0 +1,236 @@
+import gitlab
+import re
+import yaml
+import os
+import logging
+from typing import List, Optional, Set
+from urllib.parse import urlparse
+from glom import glom
+from oar.core.configstore import ConfigStore
+
+logger = logging.getLogger(__name__)
+
+class GitLabMergeRequest:
+    def __init__(self, gitlab_url: str, project_name: str, merge_request_id: int, private_token: str = None):
```

Comment:
Why do you use separate `gitlab_url` and `private_token` and not the config store?

Reply:
this class is common interface, it does not depend on config store. config store is just one option refer to `ShipmentData`. if other module what to interact with gitlab MR, it can choose any other method to get required configs.